packages/sync-service/lib/electric/schema.ex

```diff
@@ -1,10 +1,31 @@
 defmodule Electric.Schema do
   import Bitwise
+  alias Electric.Replication.Eval.Env.BasicTypes
+  alias Electric.Postgres.Inspector
+
+  @type column_name :: String.t()
+  @type type_name :: String.t()
+  @type schema :: %{
+          type: type_name(),
+          dims: non_neg_integer(),
+          max_length: non_neg_integer() | nil,
+          length: non_neg_integer() | nil,
+          precision: non_neg_integer() | nil,
+          scale: non_neg_integer() | nil,
+          fields: String.t() | nil
+        }
```

Comment:
`%{fields: String.t() | nil}` is not the same as `%{optional(:fields) => String.t()}` - latter typechecks as possible absence of a key (but not-nil if present), while former means the key is always there, but the value can be nil.
Since you're not ensuring scale/precision keys on every single column, they should use the latter form